app/models/order_updater.rb

```diff
@@ -84,11 +84,8 @@ def failed_payments?
   # shipments using a shipping method that doesn't require address, such us
   # a pickup.
   def shipping_address_from_distributor
-    shipments.each do |shipment|
-      shipping_method = shipment.shipping_method
-      next if shipping_method.require_ship_address
+    return if order.shipping_method.blank? || order.shipping_method.require_ship_address
 
-      order.ship_address = order.distributor.address
-    end
+    order.ship_address = order.__send__(:address_from_distributor)
```

Comment:
Shouldn't we make that method public if we use it?

Reply:
yeah, address_from_distributor is also a candidate for a nice little service that could also help below in shipping_address_from_distributor

Reply:
I've made this method public as well. :slightly_smiling_face: